blockchain/chaingen/generator.go

```diff
@@ -2473,6 +2515,32 @@ func (g *Generator) AssertBlockVersion(expected int32) {
 	}
 }
 
+// determineTxType determines the type of stake transaction a transaction is; if
+// none, it returns that it is an assumed regular tx.
+func (g *Generator) determineTxType(tx *wire.MsgTx) stake.TxType {
+	if isTicketPurchaseTx(tx) {
+		return stake.TxTypeSStx
+	}
+	if isVoteTx(tx) {
+		return stake.TxTypeSSGen
+	}
+	if isRevocationTx(tx) {
+		return stake.TxTypeSSRtx
+	}
+	return stake.TxTypeRegular
+}
+
+// AssertBlockVersion panics if the current tip block associated with the
+// generator does not have the specified version.
+func (g *Generator) AssertBlockRevocationTx(b *wire.MsgBlock, txIndex uint32) {
+	txType := g.determineTxType(b.STransactions[txIndex])
```

Comment:
As alluded to elsewhere, this should just be:

```Go
	if !isRevocationTx(b.STransactions[txIndex]) {
		panic(fmt.Sprintf("stake transaction at index %d in block %q is "+
			"not a revocation transaction", txIndex, g.tipName))
	}
```

Notice that I also change `with index` -> `at index`.